except_star.md

```diff
@@ -124,103 +116,161 @@ except *OSerror as errors:
+ExceptionGroup(
+  KeyError('e'),
+  ExceptionGroup(
+    TypeError('b'),
+    TypeError('c')
+  )
+)
+```
 
 Basically, before interpreting `except *` clauses, the interpreter will
-have an exception group object with a list of exceptions in it. Every
-`except *` clause, evaluated from top to bottom, can filter some of the
-exceptions out of the group and process them. In the end, if the exception
-group has no exceptions left in it, it wold mean that all exceptions were
-processed. If the exception group has some unprocessed exceptions, the current
-frame will be "pushed" to the group's traceback and the group would be
-propagated up the stack.
+have an "incoming" `ExceptionGroup` object with a list of exceptions in it
+to handle, and then:
+
+* A new empty "result" `ExceptionGroup` would be created by the interpreter.
+
+* Every `except *` clause, run from top to bottom, can filter some of the
+  exceptions out of the group and process them. If the except block crashes
+  with an error, that error is put to the "result" `ExceptionGroup` (with the
+  group of unprocessed exceptions referenced via the `__context__` attribute.)
+
+* After there are no more `except*` clauses to evaluate, there are the
+  following possibilities:
+
+  * Both "incoming" and "result" `ExceptionGroup` are empty. This means
+    that all exceptions were processed and silenced successfully.
+
+  * Both "incoming" and "result" `ExceptionGroup` are not empty.
+    This means that not all of the exceptions were matched, and some were
+    matched but either triggered new errors, or were re-raised. The interpreter
+    would merge both groups into one group and raise it.
+
+  * The "incoming" `ExceptionGroup` is non-empty: not all exceptions were
+    processed. The interpreter would raise the "incoming" group.
+
+  * The "result" `ExceptionGroup` is non-empty: all exceptions were processed,
+    but some were re-raised or caused new errors. The interpreter would
+    raise the "result" group.
+
+The order of `except*` clauses is significant just like with the regular
+`try..except`, e.g.:
+
+```python
+try:
+  raise BlockingIOError
+except *OSError as e:
+  # Would catch the error
+  print(e)
+except *BlockingIOError:
+  # Would never run
+  print('never')
+
+# would print:
+#
+#   ExceptionGroup(BlockingIOError())
+```
 
-#### Exception Chaining
+### Exception Chaining
 
-If an error occur during processing a set of exceptions in a `except *` block,
-all matched errors would be put in a new `ExceptionGroup` which would have its
-`__context__` attribute set to the just occurred exception:
+If an error occurs during processing a set of exceptions in a `except *` block,
+all matched errors would be put in a new `ExceptionGroup` which would be
+references from the just occurred exception via its `__context__` attribute:
 
 ```python
 try:
-  raise *(ValueError('a'), ValueError('b'), TypeError('z'))
+  raise ExceptionGroup(ValueError('a'), ValueError('b'), TypeError('z'))
```

Comment:
Did we discuss whether the constructor takes a list of exceptions or uses varargs (`*args`)?

Reply:
No, not yet, I believe. I guess we need to open a new issue to discuss the ExceptionGroup API in detail and later fold it into this proposal. @iritkatriel?

Reply:
I think varargs is nice. In reality I think you would want to create it empty and add exceptions to it. But whichever way it's constructed - what happens when you try to raise an empty ExceptionGroup?

Reply:
> I think varargs is nice. In reality I think you would want to create it empty and add exceptions to it. But whichever way it's constructed - what happens when you try to raise an empty ExceptionGroup?

I've created an issue to discuss that: https://github.com/python/exceptiongroups/issues/6